remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt

```diff
@@ -1,84 +1,42 @@
-//    var longitude by Delegates.notNull<Double>()
-//    var latitude by Delegates.notNull<Double>()
-//    var limit by Delegates.notNull<Int>()
-//    var level by Delegates.notNull<Int>()
-//    lateinit var host: String
-//    lateinit var userId: String
-//
-//    override fun toString() =
-//        "id: $id, name: $name, description: $description, data: $date, updated: $updated, address: $address, longitude: $longitude, latitude: $latitude, limit: $limit, level: $level, host: $host, userId: $userId"
-//
-//    fun toMap() = hashMapOf("name" to name,
-//                            "name" to name,
-//                            "description" to description,
-//                            "host" to host,
-//                            "date" to date,
-//                            "updated" to updated,
-//                            "address" to address,
-//                            "latitude" to latitude,
-//                            "longitude" to longitude,
-//                            "limit" to limit,
-//                            "level" to level,
-//                            "userId" to userId)
-//
-//    fun copy(
-//        id: String? = null,
-//        name: String? = null,
-//        description: String? = null,
-//        date: Long? = null,
-//        updated: Long? = null,
-//        address: String? = null,
-//        longitude: Double? = null,
-//        latitude: Double? = null,
-//        limit: Int? = null,
-//        level: Int? = null,
-//        host: String? = null,
-//        userId: String? = null
-//    ) = PlayoneModel().also {
-//        it.id = id ?: this.id
-//        it.name = name ?: this.name
-//        it.description = description ?: this.description
-//        it.date = date ?: this.date
-//        it.updated = updated ?: this.updated
-//        it.address = address ?: this.address
-//        it.longitude = longitude ?: this.longitude
-//        it.latitude = latitude ?: this.latitude
-//        it.limit = limit ?: this.limit
-//        it.level = level ?: this.level
-//        it.host = host ?: this.host
-//        it.userId = userId ?: this.userId
-//    }
-//}
-
 data class PlayoneModel(
-    var id: String = "",
-    var name: String = "",
-    var description: String = "",
-    var date: Long = -1L,
-    var updated: Long = -1L,
-    var address: String = "",
-    var longitude: Double = -1.0,
-    var latitude: Double = -1.0,
-    var limit: Int = -1,
-    var level: Int = -1,
-    var host: String = "",
-    var userId: String = ""
+    var id: String = defaultStr,
+    var name: String = defaultStr,
+    var description: String = defaultStr,
+    var date: Long = defaultLong,
+    var updated: Long = defaultLong,
+    var address: String = defaultStr,
+    var longitude: Double = defaultDouble,
+    var latitude: Double = defaultDouble,
+    var limit: Int = defaultInt,
+    var level: Int = defaultInt,
+    var host: String = defaultStr,
+    var userId: String = defaultStr
```

Comment:
Here is my original test code:
```
data class MockDataClass(var _abc: Int = INVALID_INT) {
    var abc by NotValidVar(_abc)
}
```

But maybe I was not writing precisely so made you confused. This is one is better,

```
data class PlayoneModel(
    private var _id: String = defaultStr,
    private var _name: String = defaultStr,
    private var _description: String = defaultStr,
    private var _date: Long = defaultLong,
    private var _updated: Long = defaultLong,
    private var _address: String = defaultStr,
    private var _longitude: Double = defaultDouble,
    private var _latitude: Double = defaultDouble,
    private var _limit: Int = defaultInt,
    private var _level: Int = defaultInt,
    private var _host: String = defaultStr,
    private var _userId: String = defaultStr
) {
    var id by NotValidVar(if (_id != defaultStr) _id else defaultStr)
    var name by NotValidVar(if (_name != defaultStr) _name else defaultStr)
    var description by NotValidVar(if (_description != defaultStr) _description else defaultStr)
    var date by NotValidVar(if (_date != defaultLong) _date else defaultLong)
    var updated by NotValidVar(if (_updated != defaultLong) _updated else defaultLong)
    var address by NotValidVar(if (_address != defaultStr) _address else defaultStr)
    var longitude by NotValidVar(if (_longitude != defaultDouble) _longitude else defaultDouble)
    var latitude by NotValidVar(if (_latitude != defaultDouble) _latitude else defaultDouble)
    var limit by NotValidVar(if (_limit != defaultInt) _limit else defaultInt)
    var level by NotValidVar(if (_level != defaultInt) _level else defaultInt)
    var host by NotValidVar(if (_host != defaultStr) _host else defaultStr)
    var userId by NotValidVar(if (_userId != defaultStr) _userId else defaultStr)
}
```

I guess you would feel it is super troublesome and actually it is using 2x memory size. I will find a better way to solve it.